src/core/network.ts

```diff
@@ -147,21 +147,21 @@ async function _networkSwitch(
 }
 
 export async function enforceNetwork(
-  provider: Provider,
-  walletClient: WalletClient,
-  switchNetwork: (chainId: number | bigint) => Promise<Chain | undefined>,
+  chainId: bigint,
+  walletClient: any,
```

Comment:
Usage of cast to WalletClient to create local instance _walletClient on line 156 starts here. 

Ideal Cleanup
- Type **any**-> **WalletClient**
- Remove use of _walletClient unless it specifically uses a local scoped version?